### PR TITLE
Aligned memory access

### PR DESCRIPTION
**Pull request by MushMal**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by MushMal** [comment]
This is still a WIP as need to be tested on kernel trap

**Comment by Sean-Der** [comment]
@MushMal anyway you know of that we can test this in travis (to make sure we don't regress?)

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h [hunk 1]
```diff
@@ -432,8 +432,8 @@ typedef VOID (*RtcOnMessage)(UINT64, BOOL, PBYTE, UINT32);
  *
  * https://www.w3.org/TR/webrtc/#dom-rtcpeerconnection-ondatachannel
  */
-typedef struct __RtcDataChannel RtcDataChannel;
-typedef VOID (*RtcOnDataChannel)(UINT64, RtcDataChannel*);
+struct __RtcDataChannel;
```
Comment (MushMal) [inline src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]: Why these changes? If we don't typedef then we need to use struct keyword in every place of usage

File: src/source/Ice/IceAgent.h [hunk 2]
```diff
@@ -151,7 +151,7 @@ typedef struct {
 
     UINT32 foundationCounter;
 
-    PTurnConnection pTurnConnection;
+    struct __TurnConnection* pTurnConnection;
```
Comment (MushMal) [inline src/source/Ice/IceAgent.h]: same

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h [hunk 3]
```diff
@@ -432,8 +432,8 @@ typedef VOID (*RtcOnMessage)(UINT64, BOOL, PBYTE, UINT32);
...
+typedef VOID (*RtcOnDataChannel)(UINT64, struct __RtcDataChannel*);
```
Comment (MushMal) [inline src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]: same

File: src/source/Ice/IceUtils.h [hunk 4]
```diff
@@ -34,7 +34,7 @@ STATUS iceUtilsGenerateTransactionId(PBYTE, UINT32);
 
 // Stun packaging and sending functions
 STATUS iceUtilsPackageStunPacket(PStunPacket, PBYTE, UINT32, PBYTE, PUINT32);
-STATUS iceUtilsSendStunPacket(PStunPacket, PBYTE, UINT32, PKvsIpAddress, PSocketConnection, PTurnConnection, BOOL);
+STATUS iceUtilsSendStunPacket(PStunPacket, PBYTE, UINT32, PKvsIpAddress, PSocketConnection, struct __TurnConnection*, BOOL);
```
Comment (MushMal) [inline src/source/Ice/IceUtils.h]: same for here and all the instances where we need to use struct keyword

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h (hunk 1 above)
Comment (MushMal) [reply src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]: Could you please describe which compilers don't do forward ref? Every one of the compilers should handle it

Comment (zhiyua-git) [reply src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]: https://stackoverflow.com/questions/8367646/redefinition-of-typedef
This doesn’t work for pure C.

Comment (MushMal) [reply src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]: I believe it's about type re-declaration and not forward decl

Comment (zhiyua-git) [reply src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]: It is the forwarding typedef. We had same issue on C producer on the old gcc compiler before.
